modules/table/src/main/java/org/apache/ignite/internal/table/TableImpl.java

```diff
@@ -371,6 +371,14 @@ private void awaitIndexes() {
         allOf(toWait.toArray(CompletableFuture[]::new)).join();
     }
 
+    /**
+     * Prepares this table for being closed. Currently, this means cancellation of PK future so that RAFT state machine
+     * does not need to wait for PK to be initialized, which allows to avoid a deadlock on stop.
+     */
+    public void prepareToClose() {
```

Comment:
I would suggest to call it `beforeClose`

Reply:
Why do we mention RAFT here?
Why do we mention a deadlock? It just releasing resources in safe way.

Reply:
Removed the specifics